### PR TITLE
Upgrade code generation to Autorest 2.0

### PR DESCRIPTION
**Pull request by RikkiGibson**
- Extracted specs out to api-specs.json like in Java.
- Using azure-rest-api-specs:current instead of master
- Specifying README.md config files for input to AutoRest. Mostly did this by copying the "source" and "args" properties from the corresponding objects in the Java api-specs.json.
- Enabled specifying a custom autorest.csharp path and added --debug flag.

### CODE REVIEW

**Comment by hovsepm** [comment]
Please remove package-lock.json file from this PR.

**Comment by RikkiGibson** [comment]
OK. Can I .gitignore package-lock.json if we don't want it committed?

**Comment by hovsepm** [comment]
@RikkiGibson yeah, you can add that file to .gitignore.

**Comment by hovsepm** [comment]
@azuresdkci retest this please

**Comment by hovsepm** [comment]
waiting for @milismsft to sign off.

File: src/ResourceManagement/api-specs.json [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    "dir": "ContainerInstance",
+    "source": "specification/containerinstance/resource-manager/readme.md",
+    "package": "Microsoft.Azure.Management.ContainerInstance.Fluent",
+    "args": "--payload-flattening-threshold=1 --tag=package-2017-08-preview"
+  },
+  "containerregistry": {
+    "dir": "ContainerRegistry",
+    "source": "specification/containerregistry/resource-manager/readme.md",
+    "package": "Microsoft.Azure.Management.ContainerRegistry.Fluent",
+    "args": "--payload-flattening-threshold=1 --tag=package-2017-03"
```
Comment (milismsft) [inline src/ResourceManagement/api-specs.json]: we should use the latest here, package-2017-10

**Pull request merged**